desktop-widgets/configuredivecomputerdialog.cpp

```diff
@@ -1445,12 +1445,12 @@ void ConfigureDiveComputerDialog::on_DiveComputerList_currentRowChanged(int curr
 		return;
 	}
 
-	int dcType = DC_TYPE_SERIAL;
+	unsigned int transport = DC_TRANSPORT_SERIAL;
 
 
 	if (selected_vendor == QString("Uemis"))
```

Comment:
I assume that the reason we don't need to hardcode "Garmin" here is that libdivecomputer will do the right thing for us here?

Reply:
No, it's actually simpler than that - only Uemis supports configuration.

Reply:
That's also why that configuredivecomputerdialog.cpp has the hardcoded transports: we don't support configuration in the general case.

This code would all go away if we made the configuration code be a button in the download window instead. Right now we duplicate a lot of the download setup and computer choice in the configuration window. Just sayin'

Reply:
I implemented that. And got rather strong pushback, especially from @glance-

Reply:
We don't support configuring Uemis computers, and thats rather just a copy paste thingie from when the initial code was written as a part of gsoc2014. All those traces of Uemis can be removed.

It's not all that much that could be removed on a merger. It was way more right after gsoc2014, but I've cleaned it up quite a bit.

In #1571 , My intention was to rather point out which issues I saw with the proposed code, and what was left and needed to be done to have a sane flow.

Reply:
TBH, the feedback felt much more like "this is my code, I use it all the time, you're making this harder". But maybe I was just in a sensitive mood when I read the comments. Either way, I agree with Linus and I continue not to like having that "configure dive computer" menu setting - but that is exactly what you appear to WANT to keep because it's what's in the video and in the guide and it's what you use. And I didn't feel like fighting over it.